modules/debug_cmd/debug_cmd.c

```diff
@@ -82,6 +82,39 @@ static int cmd_ua_debug(struct re_printf *pf, void *unused)
 }
 
 
+/**
+ * Returns all the User-Agents and their general codec state.
+ * Formatted as JSON, for use with TCP / MQTT API interface.
+ *
+ * @return All User-Agents available, NULL if none
+ */
+static int cmd_api_uastate(struct re_printf *pf, void *unused)
+{
+	(void)unused;
+	int err;
+	struct odict *od = NULL;
+	struct le *le;
+
+	err = odict_alloc(&od, 8);
+	if (err)
+		return err;
+
+	for (le = list_head(uag_list()); le && !err; le = le->next) {
+		const struct ua *ua = le->data;
+
+		err |= ua_state_json_api(od, ua);
+		/* todo: add to list */
+		err |= json_encode_odict(pf, od);
```

Comment:
this logic will not work. it will append the ua object to the dict and then print it.
if you have 3 uas the result will be:

1
1,2
1,2,3

instead you should build an array, and then encode it as the last step